### ISSUE TITLE
Panic on not common input

### ISSUE THREAD

**Issue by Stranger6667** [fancy-regex/fancy-regex#76]
Hi! Thank you for this library!

While fuzzing the [jsonschema](https://github.com/Stranger6667/jsonschema-rs/issues/253) crate, I found that `fancy-regex` panics on `\\u` (and a few more similar patterns like `\\U` and `\\x`), with:

```
thread 'main' panicked at 'index out of bounds: the len is 2 but the index is 2', /home/stranger6667/.cargo/registry/src/github.com-1ecc6299db9ec823/fancy-regex-0.6.0/src/parse.rs:394:17
```

**Code to reproduce**:

```rust
fn main() {
    fancy_regex::Regex::new("\\u");
}
```

Though it is an incomplete escape sequence, it will be nice to return `Err` in such cases as in `regex`:

```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
regex parse error:
    \u
      ^
error: incomplete escape sequence, reached end of pattern prematurely
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Probably, the solution will be to check the bounds before going to `bytes[ix]` inside `Parser::parse_hex`.
Let me know what do you think, I'll be happy to submit a PR to fix this :)

**Comment by robinst** [comment]
Oh nice, thanks for doing that :)! Yeah the fix sounds good, looking forward to the PR.

**Comment by Stranger6667** [comment]
You're very welcome :) Here goes a PR - #78

**Comment by robinst** [comment]
Released in 0.7.1 now: https://github.com/fancy-regex/fancy-regex/blob/main/CHANGELOG.md#071---2021-07-29

Let me know if you find any other problems using fuzzing!
